source/Halibut.Tests/CancellationViaClientProxyFixture.cs

```diff
@@ -68,6 +83,50 @@ public async Task CanTalkToOldServicesWhichDontKnowAboutHalibutProxyRequestOptio
                 res.Should().Be("Hello!!...");
             }
         }
+        
+        
+        [Test]
+        [LatestClientAndLatestServiceTestCases(testNetworkConditions: false)]
+        [LatestClientAndPreviousServiceVersionsTestCases(testNetworkConditions: false)]
+        [FailedWebSocketTestsBecomeInconclusive]
+        public async Task HalibutProxyRequestOptions_CanNotCancel_InFlightRequests(ClientAndServiceTestCase clientAndServiceTestCase)
+        {
+            using (var clientAndService = await clientAndServiceTestCase.CreateTestCaseBuilder()
+                       .WithStandardServices()
+                       .Build(CancellationToken))
+            {
+                var lockService = clientAndService.CreateClient<ILockService, IClientLockService>();
+
+                var cts = new CancellationTokenSource();
+                using var tmpDir = new TemporaryDirectory();
+                var fileThatOnceDeletedEndsTheCall = tmpDir.CreateRandomFile();
+                var callStartedFile = tmpDir.RandomFileName();
+
+                var takeLockASecondTime = Task.Run(() => lockService.WaitForFileToBeDeleted(fileThatOnceDeletedEndsTheCall, callStartedFile, new HalibutProxyRequestOptions(CancellationToken.None)));
+
+                Logger.Information("Waiting for the RPC call to be inflight");
+                while (!File.Exists(callStartedFile))
+                {
+                    await Task.Delay(TimeSpan.FromMilliseconds(100), CancellationToken);
+                }
+                
+                // The second call is now in flight.
```

Comment:
"The seconds call"!! There is only one call made in this test

Reply:
Old comment, ugh the number of rewrites on this test to make it work for polling and listening!

Reply:
Also `takeLockASecondTime` could use a rename

Reply:
renamed.